contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1422 @@
+    function setNotificationReward(uint96 reward) external onlyGovernance {
+        notificationReward = reward;
+        emit NotificationRewardSet(reward);
+    }
+
+    /// @notice Transfer some amount of T tokens as reward for notifications
+    ///         of misbehaviour
+    function pushNotificationReward(uint96 reward) external {
+        require(reward > 0, "Reward must be specified");
+        notifiersTreasury += reward;
+        emit NotificationRewardPushed(reward);
+        token.safeTransferFrom(msg.sender, address(this), reward);
+    }
+
+    /// @notice Adds operators to the slashing queue along with the amount that
+    ///         should be slashed from each one of them. Can only be called by
+    ///         application authorized for all operators in the array.
+    /// @dev    This method doesn't emit events for operators that are added to
+    ///         the queue. If necessary  events can be added to the application
+    ///         level.
+    function slash(uint96 amount, address[] memory _operators)
+        external
+        override
+    {
+        notify(amount, 0, address(0), _operators);
+    }
+
+    /// @notice Adds operators to the slashing queue along with the amount.
+    ///         The notifier will receive reward per each operator from
+    ///         notifiers treasury. Can only be called by application
+    ///         authorized for all operators in the array.
+    /// @dev    This method doesn't emit events for operators that are added to
+    ///         the queue. If necessary  events can be added to the application
+    ///         level.
+    function seize(
+        uint96 amount,
+        uint256 rewardMultiplier,
+        address notifier,
+        address[] memory _operators
+    ) external override {
+        notify(amount, rewardMultiplier, notifier, _operators);
+    }
+
+    /// @notice Takes the given number of queued slashing operations and
+    ///         processes them. Receives 5% of the slashed amount.
+    ///         Executes `involuntaryAuthorizationDecrease` function on each
+    ///         affected application.
+    function processSlashing(uint256 count) external override {
+        require(
+            slashingQueueIndex < slashingQueue.length && count > 0,
+            "Nothing to process"
+        );
+
+        uint256 maxIndex = slashingQueueIndex + count;
+        maxIndex = Math.min(maxIndex, slashingQueue.length);
+        count = maxIndex - slashingQueueIndex;
+        uint96 tAmountToBurn = 0;
+
+        uint256 index = slashingQueueIndex;
+        for (; index < maxIndex; index++) {
+            SlashingEvent storage slashing = slashingQueue[index];
+            tAmountToBurn += processSlashing(slashing);
+        }
+        slashingQueueIndex = index;
+
+        uint256 tProcessorReward = uint256(tAmountToBurn).percent(
+            SLASHING_REWARD_PERCENT
+        );
+        notifiersTreasury += tAmountToBurn - tProcessorReward.toUint96();
```

Comment:
‼️ I think we should add a function allowing the governance to burn (or transfer out) the given amount from `notifiersTreasury`. Tokens landing there are not actually burned and I can imagine situation when DAO might want it to happen. One example is the discussion we are having in Keep community now about burning some KEEP unspent for network incentives to improve the token ratio for KEEP<>T and making KEEP holders happier.

Reply:
✔️